src/wallet/wallet_rpc_server.cpp

```diff
@@ -2522,20 +2507,19 @@ bool wallet_rpc_server::on_open_wallet(const wallet_rpc::COMMAND_RPC_OPEN_WALLET
 	try
 	{
 		wal = tools::wallet2::make_from_file(vm2, wallet_file, nullptr).first;
+		if(!wal)
+		{
+			er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+			er.message = "Failed to open wallet";
+			return false;
+		}
+
+		start_wallet_backend(std::move(wal));
 	}
 	catch(const std::exception &e)
 	{
 		handle_rpc_exception(std::current_exception(), er, WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR);
```

Comment:
IMO a `return false` is missing if we catch an exeption

Reply:
Nice catch